src/scripts/main.js

```diff
@@ -1 +1,82 @@
 'use strict';
+
+let leftClicked = false;
+let rightClicked = false;
+
+function createMessage(message, className) {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.classList.add(className);
+  div.textContent = message;
+  document.body.append(div);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    reject(new Error(`First promise was rejected`));
+  }, 3000);
+
+  document.addEventListener(
+    'click',
+    () => {
+      leftClicked = true;
+
+      resolve('First promise was resolved');
+    },
+    { once: true },
+  );
+});
+
+firstPromise
+  .then((data) => {
+    createMessage(data, 'success');
+  })
+  .catch((error) => {
+    createMessage(error, 'error');
+  });
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener(
+    'click',
+    () => {
+      leftClicked = true;
+
+      resolve('Second promise was resolved');
+    },
+    { once: true },
+  );
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    rightClicked = true;
+
+    resolve('Second promise was resolved');
+  });
+});
```

Comment:
The second promise resolves twice: once for a left click and once for a right click. This is not ideal as it should resolve only once on any click. Consider using a single event listener that resolves the promise on the first click, regardless of the button used.